src/helpers/serviceSelector.ts

```diff
@@ -0,0 +1,74 @@
+import { ProcessorTransformationRequest, RouterTransformationRequestData } from '../types/index';
+import { INTEGRATION_SERVICE } from '../routes/utils/constants';
+import CDKV1DestinationService from '../services/destination/cdkV1Integration';
+import CDKV2DestinationService from '../services/destination/cdkV2Integration';
+import DestinationService from '../interfaces/DestinationService';
+import NativeIntegrationDestinationService from '../services/destination/nativeIntegration';
+import SourceService from '../interfaces/SourceService';
+import NativeIntegrationSourceService from '../services/source/nativeIntegration';
+
+export default class ServiceSelector {
+  private static serviceMap: Map<string, any> = new Map();
+
+  private static isCdkDestination(destinationDefinitionConfig: Object) {
+    return !!destinationDefinitionConfig?.['cdkEnabled'];
+  }
+
+  private static isCdkV2Destination(destinationDefinitionConfig: Object) {
+    return !!destinationDefinitionConfig?.['cdkV2Enabled'];
+  }
+
+  private static fetchCachedService(serviceType: string) {
+    let service: any;
+    if (this.serviceMap.has(serviceType)) {
+      service = this.serviceMap.get(serviceType);
```

Comment:
use return stmt here so that we can rewrite else block without else

Reply:
updated